### PR TITLE
Release/0.1.16

### PR DESCRIPTION
**Pull request by cortadocodes**
### Breaking changes
- [x] Rename `Service.__init__` parameter `id` to `service_id` to avoid built-in name clash
- [x] Move `deployment` package into `cloud` package

### Dependencies
- [x] Use newest version of `twined` to support python>=3.6

### Minor improvements
- [x] Remove duplicate code and unnecessary comments from `Runner`
- [x] Raise error if `SERVICE_ID` envvar is missing from deployment environment
- [x] Disallow non-None empty values as `Service` IDs 
- [x] Add base class for service backends; update docstrings

### Fixes
- [x] Use `OctueJSONEncoder` in JSON serialisation inside `Service.answer` to ensure `numpy` arrays are serialised

### Testing
- [x] Add tests for `Topic` and `Subscription`
- [x] Add extra test for `Service`
- [x] Shorten runtime of `cli.start` test

### CODE REVIEW

File: octue/runner.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ class Runner:
     The Runner class provides a set of configuration parameters for use by your application, together with a range of
     methods for managing input and output file parsing as well as controlling logging.
 
-    :param str twine: string path to the twine file, or a string containing valid twine json
+    :param str|twined.Twine twine: string path to the twine file, or a string containing valid twine json
```
Comment (thclark) [inline octue/runner.py:26]: updated type but not description

File: octue/runner.py [hunk 2]
```diff
@@ -57,15 +57,13 @@ def __init__(
...
-
-        # Store the log level (same log level used for all analyses)
         self._log_level = log_level
         self.handler = handler
 
         if show_twined_logs:
             apply_log_handler(logger=package_logger, handler=self.handler)
 
-        # Ensure the twine is present and instantiate it
+        # Ensure the twine is present and instantiate it.
```
Comment (thclark) [inline octue/runner.py:66]: I prefer to leave full stops off unless it's a multi-sentence comment. Happy to change but we should decide on a style and note it.

**Review by thclark — APPROVED** [review]
Two minor tweaks only ;)

File: octue/runner.py (hunk 2 above)
Comment (cortadocodes) [reply octue/runner.py:66]: Ah cool. I think it's more consistent to end all comments with full stops? Although in docstring `:param` descriptions, I prefer to leave them off for some reason?

**Pull request merged**